components/camel-http4/src/test/java/org/apache/camel/component/http4/HttpProducerContentTypeWithSemiColomnTest.java

```diff
@@ -0,0 +1,113 @@
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.http4;
+
+import java.io.IOException;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.http.HttpException;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.bootstrap.HttpServer;
+import org.apache.http.impl.bootstrap.ServerBootstrap;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.protocol.HttpRequestHandler;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class HttpProducerContentTypeWithSemiColomnTest extends BaseHttpTest {
+
+    private static final String CONTENT_TYPE = "multipart/form-data;boundary=---------------------------j2radvtrk";
+    
+    private HttpServer localServer;
+    
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        localServer = ServerBootstrap.bootstrap().
+                setHttpProcessor(getBasicHttpProcessor()).
+                setConnectionReuseStrategy(getConnectionReuseStrategy()).
+                setResponseFactory(getHttpResponseFactory()).
+                setExpectationVerifier(getHttpExpectationVerifier()).
+                setSslContext(getSSLContext()).
+                registerHandler("/content", new HttpRequestHandler() {
+                    @Override
+                    public void handle(HttpRequest request, HttpResponse response, HttpContext context) throws HttpException, IOException {
+                        String contentType = request.getFirstHeader(Exchange.CONTENT_TYPE).getValue();
+                        
+                        assertEquals(CONTENT_TYPE.replace(";", "; "), contentType);
+
+                        response.setEntity(new StringEntity(contentType, "ASCII"));
+                        response.setStatusCode(HttpStatus.SC_OK);
+                    }
+                }).create();
+        localServer.start();
+
+        super.setUp();
+    }
+
+    @After
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+
+        if (localServer != null) {
+            localServer.stop();
+        }
+    }
+    
+    @Test
+    public void testContentTypeWithBoundary() throws Exception {
+        Exchange out = template.request("http4://" + localServer.getInetAddress().getHostName() + ":" + localServer.getLocalPort() + "/content", new Processor() {
+
+            @Override
+            public void process(Exchange exchange) throws Exception {
+                exchange.getIn().setHeader(Exchange.CONTENT_TYPE, CONTENT_TYPE);
+                exchange.getIn().setBody("This is content");
+            }
+            
+        });
+
+        assertNotNull(out);
+        assertFalse("Should not fail", out.isFailed());
+        assertEquals(CONTENT_TYPE.replace(";", "; "), out.getOut().getBody(String.class));
```

Comment:
these two lines are tricky because in request the header is caught in handler with space even if you don't provide it.